Weather/WeatherViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  WeatherViewModel.swift
+//  Weather
+//
+//  Created by Serena Roderick on 9/13/24.
+//
+
+import UIKit
+protocol WeatherViewModelProtocol {
+    
+    func getApiData() throws
+    func handleApiData()
+    func passDataToView(data: Data)
+    func kelvinToCelcius(kelvin: Double) -> Double
+    func kelvinToFarenheit(kelvin: Double) -> Double
+}
+
+class WeatherViewModel: WeatherViewModelProtocol {
+
+    var weatherData: [WeatherData]?
+    var networkManager: NetworkManager?
```

Comment:
is this mutlabe, can we use let ,    insert this as mandtory depdency using intilizer